codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -736,148 +757,142 @@ public Character getCharAndRemove(T name) {
     @Override
     public char getCharAndRemove(T name, char defaultValue) {
         Character v = getCharAndRemove(name);
-        return v == null ? defaultValue : v;
+        return v != null ? v : defaultValue;
     }
 
     @Override
     public Short getShortAndRemove(T name) {
         T v = getAndRemove(name);
-        if (v == null) {
-            return null;
-        }
-        try {
-            return valueConverter.convertToShort(v);
-        } catch (Throwable ignored) {
-            return null;
-        }
+        return v != null ? valueConverter.convertToShort(v) : null;
     }
 
     @Override
     public short getShortAndRemove(T name, short defaultValue) {
         Short v = getShortAndRemove(name);
-        return v == null ? defaultValue : v;
+        return v != null ? v : defaultValue;
     }
 
     @Override
     public Integer getIntAndRemove(T name) {
         T v = getAndRemove(name);
-        if (v == null) {
-            return null;
-        }
-        try {
-            return valueConverter.convertToInt(v);
-        } catch (Throwable ignored) {
-            return null;
-        }
+        return v != null ? valueConverter.convertToInt(v) : null;
```

Comment:
Is it documented, that if an Exception is thrown the value is still removed? That might be confusing else.